tests/acceptance/features/bootstrap/Sharing.php

```diff
@@ -417,35 +418,45 @@ public function userHasCreatedAShareWithSettings(string $user, ?TableNode $body)
 	}
 
 	/**
-	 * @When /^user "([^"]*)" creates a public link share using the sharing API with settings$/
-	 *
 	 * @param string $user
-	 * @param TableNode|null $body
+	 * @param TableNode $body
 	 *
-	 * @return void
-	 * @throws Exception
+	 * @return ResponseInterface
 	 */
-	public function userCreatesAPublicLinkShareWithSettings(string $user, ?TableNode $body):void {
+	public function createPublicLinkShare(string $user, TableNode $body): ResponseInterface {
 		$rows = $body->getRows();
 		// A public link share is shareType 3
 		$rows[] = ['shareType', 'public_link'];
 		$newBody = new TableNode($rows);
-		$response = $this->createShareWithSettings($user, $newBody);
-		$this->setResponse($response);
+		return $this->createShareWithSettings($user, $newBody);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" creates a public link share using the sharing API with settings$/
+	 *
+	 * @param string $user
+	 * @param TableNode $body
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userCreatesAPublicLinkShareWithSettings(string $user, TableNode $body):void {
+		$this->setResponse($this->createPublicLinkShare($user, $body));
 		$this->pushToLastStatusCodesArrays();
 	}
 
 	/**
 	 * @Given /^user "([^"]*)" has created a public link share with settings$/
 	 *
 	 * @param string $user
-	 * @param TableNode|null $body
+	 * @param TableNode $body
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function userHasCreatedAPublicLinkShareWithSettings(string $user, ?TableNode $body):void {
-		$this->userCreatesAPublicLinkShareWithSettings($user, $body);
+	public function userHasCreatedAPublicLinkShareWithSettings(string $user, TableNode $body):void {
+		$this->setResponse($this->createPublicLinkShare($user, $body));
+		$this->pushToLastStatusCodesArrays();
```

Comment:
We must not use `setResponse` is Given steps as well.
```suggestion
		$response = $this->createPublicLinkShare($user, $body);
```